lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
@gmile me and @lexmag discuss this stuff way more than we should so I definitely know about that :P, but that style guide is not relevant to Elixir's source code anyways (and `when` indentation is one of the cases where they differ). That said, I'm in the process of convincing Aleksei that indented `when` is better than non-indented `when` 😄